server/lists/listController.js

```diff
@@ -113,38 +113,41 @@ module.exports = {
     });
   },
 
-  deleteItemFromList: function(req, res) {
+  updateItem: function(req, res) {
     var username = interimUsername;
-    var itemName = req.body.name.toLowerCase();
+    var newName = req.body.name.toLowerCase();
+    var index = req.body.index;
+
+    var setModifier = { $set: {} };
+    setModifier.$set['list.' + index] = req.smartShoppingData._id;
+    User.findOneAndUpdate({username: username}, setModifier, {upsert: true}, function(err, user) {
+      if (err) {
+        console.log(err);
+        res.status(500).send({error: 'Server Error'});
+      } 
+      res.send(user.list);
+    }); 
+  },
 
+  deleteItemFromList: function(req, res) {
+    var username = interimUsername;
+    var index = Number(req.body.index);
```

Comment:
I don't think this line is necessary, since you only use `index` once on line 122 as a string.